app/middleware/renderer.js

```diff
@@ -1,26 +1,29 @@
 const messages = require('../lib/messages');
-const postcodeValidator = require('../lib/postcodeValidator');
 const log = require('../lib/logger');
 
 function results(req, res) {
-  res.render('results');
+  if (res.locals.gps.length > 0) {
+    res.render('results');
+  } else {
+    res.render('no-results');
```

Comment:
I think 0 results is a theoretical, future possibility but I don't believe there are any postcodes currently which are 1) English and 2) which return no results.

We have unit tests for the scenario.